### PR TITLE
Adding max http response string length as a setting, and capping http…

### PR DESCRIPTION
**Pull request by toepkerd**
Adding max http response string length as a setting, and capping http response string based on that setting

### Description
This is a 2.x version of the following change merged to main/3.0: https://github.com/opensearch-project/notifications/pull/861. 2.x did not have all the necessary libraries that this change needed, so they were reimplemented for the purposes of working on older versions.

### Issues Resolved
Please refer to https://github.com/opensearch-project/notifications/pull/861

### Check List
- [Y] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: notifications/core/src/main/kotlin/org/opensearch/notifications/core/client/DestinationHttpClient.kt [hunk 1]
```diff
@@ -182,4 +206,44 @@ class DestinationHttpClient {
...
+        val buf = CharArrayBuffer(minOf(maxResultLength, if (contentLength > 0) contentLength else DEFAULT_CHAR_BUFFER_SIZE))
+        val reader = InputStreamReader(inStream, actualCharSet)
+        val tmp = CharArray(DEFAULT_CHAR_BUFFER_SIZE)
+        var chReadCount: Int
+        while (reader.read(tmp).also { chReadCount = it } != -1 && buf.length < maxResultLength) {
+            buf.append(tmp, 0, chReadCount)
+        }
+        buf.setLength(minOf(buf.length, maxResultLength))
+        return buf
+    }
```
Comment (Hailong-am) [inline notifications/core/src/main/kotlin/org/opensearch/notifications/core/client/DestinationHttpClient.kt:248]: can we move this truncate logic to a util class like `EntityUtils`?

**Review by Hailong-am — APPROVED** [review]
LGTM, thanks

File: notifications/core/src/main/kotlin/org/opensearch/notifications/core/setting/PluginSettings.kt [hunk 2]
```diff
@@ -482,6 +510,11 @@ internal object PluginSettings {
             log.debug("$LOG_PREFIX:$SOCKET_TIMEOUT_MILLISECONDS_KEY -autoUpdatedTo-> $clusterSocketTimeout")
             socketTimeout = clusterSocketTimeout
         }
+        val clusterMaxHttpResponseSize = clusterService.clusterSettings.get(MAX_HTTP_RESPONSE_SIZE)
+        if (clusterMaxHttpResponseSize != null) {
+            log.debug("$LOG_PREFIX:$MAX_HTTP_RESPONSE_SIZE_KEY -autoUpdatedTo-> $clusterMaxHttpResponseSize")
+            socketTimeout = clusterSocketTimeout
```
[the remark this answers is not in the document]
Comment (toepkerd) [reply notifications/core/src/main/kotlin/org/opensearch/notifications/core/setting/PluginSettings.kt]: fixed

File: notifications/core/src/main/kotlin/org/opensearch/notifications/core/client/DestinationHttpClient.kt (hunk 1 above)
Comment (toepkerd) [reply notifications/core/src/main/kotlin/org/opensearch/notifications/core/client/DestinationHttpClient.kt:248]: If we do let's keep it as a backlog item

**Review by Hailong-am — APPROVED** [review]

**Review by sbcd90 — APPROVED** [review]

**Pull request merged**